test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -5707,6 +5707,27 @@ def forward(self, input, other):
         y = torch.randint(10, (5, ))
         self.run_test(MatmulModel(), (x, y))
 
+    @skipIfUnsupportedMinOpsetVersion(9)  # MatMul long inputs is added in ONNX opset 9.
+    def test_dot(self):
+        class MatmulModel(torch.nn.Module):
+            def forward(self, input, other):
+                return torch.dot(input, other)
+
+        x = torch.randn(5, requires_grad=True)
+        y = torch.randn(5, requires_grad=True)
+        self.run_test(MatmulModel(), (x, y))
+
+        x = torch.randint(10, (5, ))
+        y = torch.randint(10, (5, ))
+        self.run_test(MatmulModel(), (x, y))
+
+    @disableScriptTest()  # SpectralNorm not TorchScript compatible.
+    def test_spectral_norm(self):
+        m = torch.nn.utils.spectral_norm(torch.nn.Linear(2, 4))
```

Comment:
What is `torch.nn.utils.spectral_norm` look like in torchscript? Is it composed of several Aten operators and one of them is `dot`?

Reply:
src: https://github.com/pytorch/pytorch/blob/master/torch/nn/utils/spectral_norm.py SpectralNorm is not nn.Module, just a regular python class.